From: Rob Evers <revers@redhat.com>
Date: Wed, 27 Oct 2010 15:22:54 -0400
Subject: [scsi] lpfc: fix IOCB leak on FDISC completion
Message-id: <1288192999-24221-3-git-send-email-revers@redhat.com>
Patchwork-id: 28927
O-Subject: [RHEL5.6 PATCH 02/27] lpfc: Fix for IOCB leak on FDISC completion.
	(CR: 106902)
Bugzilla: 639028

lpfc: Fix for IOCB leak on FDISC completion. (CR: 106902)

From: Rob Evers on behalf of Emulex <revers@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=636289

diff --git a/drivers/scsi/lpfc/lpfc_els.c b/drivers/scsi/lpfc/lpfc_els.c
index b264491..6a8ff16 100644
--- a/drivers/scsi/lpfc/lpfc_els.c
+++ b/drivers/scsi/lpfc/lpfc_els.c
@@ -6920,7 +6920,7 @@ lpfc_cmpl_els_fdisc(struct lpfc_hba *phba, struct lpfc_iocbq *cmdiocb,
 		 * to update the MAC address.
 		 */
 		lpfc_register_new_vport(phba, vport, ndlp);
-		return ;
+		goto out;
 	}
 
 	if (vport->fc_flag & FC_VPORT_NEEDS_INIT_VPI)
